munit/shared/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -144,80 +144,215 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
     if (PlatformCompat.isIgnoreSuite(cls)) {
       val description = getDescription()
       notifier.fireTestIgnored(description)
-      return Future.successful(())
-    }
-    var isBeforeAllRun = false
-    val result = {
-      val isContinue = runBeforeAll(notifier)
-      isBeforeAllRun = isContinue
-      if (isContinue) {
-        runAsyncTestsSynchronously(notifier)
-      } else {
-        Future.successful(())
+      Future.successful(())
+    } else {
+      val result =
+        for {
+          isBeforeAllRun <- runBeforeAll(notifier)
+          if isBeforeAllRun
```

Comment:
What is the error message when this condition fails? It would be best to fail with an exception that explains the reason why tests didn't run: "Aborting test execution because `beforeAll` hooks failed to run" (or something like that).

Reply:
Shame on me. Nice catch, though! I've rewritten it with preserving the semantic of the original version.